ktor-client/ktor-client-java/jvm/src/io/ktor/client/engine/java/JavaHttpEngine.kt

```diff
@@ -45,7 +45,14 @@ public class JavaHttpEngine(override val config: JavaHttpConfig) : HttpClientEng
 
     private var httpClient: HttpClient? = null
 
+    override val coroutineContext: CoroutineContext
+
     init {
+        val parentContext = super.coroutineContext
+        val parent = parentContext[Job]!!
```

Comment:
So now we have two jobs and only one of them is completed. Who will complete the `parent`?

Reply:
reverted, will investigate it for other engines